### PR TITLE
fix: correctly add/remove chain adapters during app launch

### PR DESCRIPTION
**Pull request by cjthompson**
## Description

fix: app crashes after last merge due to a circular dependency that only showed up in optimized production builds

feat: setup bitcoin and ethereum plugins with chain adapters

## Notice

- [X] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [X] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Certain pages may expect Ethereum/Bitcoin chain adapters to always exist and may not work correctly. There was a change to Yearn that needs to be tested.

## Testing

All the things

### CODE REVIEW

**Review by 0xdef1cafe — APPROVED** [review]

**Review by theoboldfrazier — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -35,41 +33,44 @@ export const App = () => {
...
+            if (plugin.routes) {
+              routes = routes.concat(plugin.routes)
+            }
 
             // chain adapters providers
             plugin.providers?.chainAdapters?.forEach(([chain, factory]) => {
-              chainAdapterManager.addChain(chain, factory)
               // track newly registered adapters by plugins
-              newChainAdapters.push(chain)
+              newChainAdapters[chain] = factory
```
Comment (0xdef1cafe) [inline src/App.tsx:54]: please check with switching wallets, this is not working correctly w/ metamask. i know metamask doesnt support cosmos, but the dashboard doesn't load when enabling the flag

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
please see comment

File: src/App.tsx (hunk 1 above)
Comment (cjthompson) [reply src/App.tsx:54]: the fix was in `getAccountSpecifiers`. We definitely need to move this into redux state.

Comment (gomesalexandre) [reply src/App.tsx:54]: Switched back and forth between Metamask and native wallet with this diff applied, this seems fixed

**Review by gomesalexandre — APPROVED** [review]
LGTM

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**